Source/WTF/wtf/glib/GRefPtr.h

```diff
@@ -207,7 +234,15 @@ template <typename T> struct IsSmartPtr<GRefPtr<T>> {
     static constexpr bool isNullable = true;
 };
 
-template <typename T> GRefPtr<T> adoptGRef(T* p)
+// Transfer ownership of a raw pointer non-floating reference into a new GRefPtr.
+// This involves no refcount increases or calls to glib.
+//
+// adoptGRef() must NEVER be used with floating references, as any ref_sink() would cause
+// the reference to be stolen from GRefPtr.
+//
+// Specializations for types where floating references are common should contain debug assertions
+// guarding against accidental use with floating references.
+template <typename T> GRefPtr<T> adoptGRef(T* p /* (transfer full) */)
```

Comment:
You could add here that this corresponds to transfer full.

Reply:
It already says `/* (transfer full) */`, but I could mention it in the first paragraph too if that's what you meant.